File: app/src/main/java/br/com/popularmovies/movies/viewmodel/factories/MovieFactory.java
package br.com.popularmovies.movies.viewmodel.factories;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import br.com.popularmovies.movies.viewmodel.MovieViewModel;
import br.com.popularmovies.services.movieService.MovieRepositoryImpl;

public class MovieFactory implements ViewModelProvider.Factory {
    private final MovieRepositoryImpl mMovieRepository;

    public MovieFactory(MovieRepositoryImpl mMovieRepository) {
        this.mMovieRepository = mMovieRepository;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new MovieViewModel(mMovieRepository);
    }
}

File: app/src/main/java/br/com/popularmovies/movies/viewmodel/factories/MovieFactory.kt
package br.com.popularmovies.movies.viewmodel.factories

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import br.com.popularmovies.movies.viewmodel.MovieViewModel
import br.com.popularmovies.repositories.movie.MovieRepository
import javax.inject.Inject

class MovieFactory @Inject constructor(
        private val mMovieRepository: MovieRepository
) : ViewModelProvider.Factory {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return MovieViewModel(mMovieRepository) as T
    }
}